projectile.el

```diff
@@ -3215,7 +3215,8 @@ This command will first prompt for the directory the file is in."
   (cl-mapcan
    (lambda (project)
      (when (file-exists-p project)
-       (let ((default-directory project))
+       (let ((default-directory project)
+             (projectile-cached-project-root nil))
```

Comment:
Yeah, definitely. I think the approach with dynamic vars was an epic mistake on my part and I regret taking this shortcut in the early days of the project. I would have changed this a while ago if I had any time for OSS work.